src/test/c_api_stress_tests.cpp

```diff
@@ -0,0 +1,441 @@
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiGetMetadataInALoop,
+        &StressCapiConfigChanges::retireSpecificVersionUsed,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+
+TEST_F(StressCapiConfigChanges, AddModelDuringGetModelStatusLoad) {
+    bool performWholeConfigReload = true;  // we just need to have all model versions rechecked
+    std::set<StatusCode> requiredLoadResults = {
+        StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {
+        StatusCode::MODEL_VERSION_MISSING  // this should be hit if test is stressing enough, sporadically does not happen
+    };
+    performStressTest(
+        &ConfigChangeStressTest::triggerCApiGetStatusInALoop,
+        &ConfigChangeStressTest::addFirstModel,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+
+class StressPipelineCustomNodesWithPreallocatedBuffersCapiConfigChanges : public StressCapiConfigChanges {
+public:
+    void checkInferResponse(OVMS_InferenceResponse* response) override {
+        ASSERT_NE(response, nullptr);
+        uint32_t outputCount = 42;
+        ASSERT_CAPI_STATUS_NULL(OVMS_InferenceResponseGetOutputCount(response, &outputCount));
+        ASSERT_EQ(outputCount, 1);
+        const void* voutputData = nullptr;
+        size_t bytesize = 42;
+        uint32_t outputId = 0;
+        OVMS_DataType datatype = (OVMS_DataType)199;
+        const int64_t* shape{nullptr};
+        size_t dimCount = 42;
+        OVMS_BufferType bufferType = (OVMS_BufferType)199;
+        uint32_t deviceId = 42;
+        const char* outputName{nullptr};
+        ASSERT_CAPI_STATUS_NULL(OVMS_InferenceResponseGetOutput(response, outputId, &outputName, &datatype, &shape, &dimCount, &voutputData, &bytesize, &bufferType, &deviceId));
+        ASSERT_EQ(std::string("custom_dummy_output"), outputName);
+        EXPECT_EQ(datatype, OVMS_DATATYPE_FP32);
+        EXPECT_EQ(dimCount, 2);
+        EXPECT_EQ(bufferType, OVMS_BUFFERTYPE_CPU);
+        EXPECT_EQ(deviceId, 0);
+        for (size_t i = 0; i < DUMMY_MODEL_SHAPE.size(); ++i) {
+            EXPECT_EQ(DUMMY_MODEL_SHAPE[i], shape[i]) << "Different at:" << i << " place.";
+        }
+        const float* outputData = reinterpret_cast<const float*>(voutputData);
+        ASSERT_EQ(bytesize, sizeof(float) * DUMMY_MODEL_INPUT_SIZE);
+        std::array<float, DUMMY_MODEL_INPUT_SIZE> data{0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        for (size_t i = 0; i < data.size(); ++i) {
+            EXPECT_EQ(data[i] + 2, outputData[i]) << "Different at:" << i << " place.";
+        }
+    }
+};
+
+TEST_F(StressPipelineCustomNodesWithPreallocatedBuffersCapiConfigChanges, RemoveCustomLibraryDuringPredictLoad) {
+    SetUpConfig(stressPipelineCustomNodeAddOneThenDummy);
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
```

Comment:
Why different statuses here than:
  TEST_F(StressPipelineCustomNodesWithPreallocatedBuffersConfigChanges, RemoveCustomLibraryDuringPredictLoad?

Reply:
Do we hit this status code?
:MODEL_VERSION_NOT_LOADED_ANYMORE